gulpfile.js

```diff
@@ -225,6 +225,30 @@ function buildExtensions(options) {
   return Promise.all(results);
 }
 
+/**
+ * Process the command line argument --extensions from a list of extensions
+ * and example AMP documents into a single list of extensions.
+ * @param {string} argvExtensions
+ * @return {!Array<string>}
+ */
+function getExtensionsFromArg(argvExtensions) {
+  const args = argvExtensions.split(',');
+  const extensionsToBuild = args.filter(arg => !arg.endsWith('html'));
+  const examples = args.filter(arg => arg.endsWith('html'));
+
+  examples.forEach(example => {
+    const html = fs.readFileSync(example, 'utf8');
+    const customElementTemplateRe = /custom-(element|template)="([^"]+)"/g;
+    const extensionNameMatchIndex = 2;
+    let match;
+    while ((match = customElementTemplateRe.exec(html))) {
+      extensionsToBuild.push(match[extensionNameMatchIndex]);
```

Comment:
If `examples/foo.html` and `examples/bar.html` both use the extension `amp-baz`, will this code not add it twice to `extensionsToBuild`? I think you may have to de-dupe here.

Reply:
Ah you're right, thanks for pointing that out.